source/literal-list.d.ts

```diff
@@ -0,0 +1,152 @@
+import type {IfNotAnyOrNever} from './internal/type.d.ts';
+import type {UnionToTuple} from './union-to-tuple.d.ts';
+import type {UnknownArray} from './unknown-array.d.ts';
+import type {BuildTuple} from './internal/tuple.d.ts';
+import type {Join, JoinableItem} from './join.d.ts';
+import type {JoinUnion} from './join-union.d.ts';
+import type {IsNever} from './is-never.d.ts';
+import type {IsUnion} from './is-union.d.ts';
+
+/**
+Create a tuple where each element is the union `U`, with the length equal to the number of members in `U`.
+
+@example
+```
+type T1 = TupleOfUnions<'a' | 'b'>;
+//=> ['a' | 'b', 'a' | 'b']
+
+type T2 = TupleOfUnions<1 | 2 | 3>;
+//=> [1 | 2 | 3, 1 | 2 | 3, 1 | 2 | 3]
+```
+*/
+type TupleOfUnions<U> = UnionToTuple<U>['length'] extends infer Length extends number
+	? BuildTuple<Length, U>
+	: never;
+
+/**
+Convert a tuple or union type into a string representation. Used for readable error messages in other types.
+
+ - `S`: **separator** between members (`default: ','`)
+ - `E`: **start** and **end** delimiters of the string (`default: ['', '']`)
+
+@example
+```
+type T1 = TypeAsString<['a', 'b'], ', ', ['[', ']']>;
+//=> '[a, b]'
+
+type T2 = TypeAsString<'a' | 'b', ' | '>;
+//=> 'a | b'
+```
+*/
+// TODO: Make a separate `Stringify` type `JoinableItem[]` mixed with `JoinableItem`
+type TypeAsString<T, S extends string = ',', E extends [string, string] = ['', '']> =
+	`${E[0]}${
+		[T] extends [readonly JoinableItem[]] // TODO: add `JoinableArray` type
+			? IsUnion<T> extends true
+				? JoinUnion<`[${Join<T, ', '>}]`, S>
+				: Join<T, S>
+			: [T] extends [JoinableItem]
+				? JoinUnion<T, S>
+				: '...' // To Complex
```

Comment:
Use AI to catch typos please.
```suggestion
				: '...' // Too complex
```